src/preamble.js

```diff
@@ -46,12 +46,33 @@ function getSafeHeapType(bytes, isFloat) {
 var SAFE_HEAP_COUNTER = 0;
 #endif
 
+var safe_heap_warn_set = new Set();
+
 function SAFE_HEAP_STORE(dest, value, bytes, isFloat) {
 #if SAFE_HEAP_LOG
   Module.print('SAFE_HEAP store: ' + [dest, value, bytes, isFloat, SAFE_HEAP_COUNTER++]);
 #endif
   if (dest <= 0) abort('segmentation fault storing ' + bytes + ' bytes to address ' + dest);
-  if (dest % bytes !== 0) abort('alignment error storing to address ' + dest + ', which was expected to be aligned to a multiple of ' + bytes);
+  if (dest % bytes !== 0)
```

Comment:
I see what you mean now, thanks. Yes, this does need changes in SAFE_HEAP.

I'm afraid though that that makes me less positive on merging this. I think this is a fairly niche case, and will become irrelevant by wasm, so I was hoping it was trivial to support in which case the tradeoff would be easy to make. But given the complexity here, I am leaning the other way now.

Curious to hear what other people think, though, maybe it's not as niche as I believe.